### PR TITLE
Create a contributor-guide page

### LINKED ISSUE #74: Create a contributor-guide page
Currently `README.md` page contains lot of information( Continuing from #6  ) - also contains some information related to contributors guide but its not solid. So its better to divide the `README.md` into : 

1.contributor-guide page and give a link from readme to contributor-guide page
     
   - This contains all the contents under the sub head `How to compile using Gradle?` and 
      `How do I install Susi: Download, Build, Run`.  
   
2.documentation page give a link from readme to documentation page 
   
   - This will have all the contents under the sub head `Where can I find documentation?` 

Remaining information will be there in `README.md` and it also contains links to both of of these pages.

Reasons to do so : 

* Looks good. 

* It will be easy to maintain.

* Removes unnecessary( not required in readme but in some other doc) contents from `readme.md`.

### PR DESCRIPTION
**Pull request by AnaghHegde**
Fixes #74 #6 

I have separated/refactored the readme.md into subpages with relevant information according to my knowledge. I think still some edits are required. As of now i have just divided the `readme` file.

### CODE REVIEW

File: docs/contributor-guide.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  gradle assemble
+  ```
+
+  Compiled file can be found in build dir
+  Last, clean up so that we can still build the project using Ant
+
+  ```
+  ./gradle_clean.sh
+  ```
+
```
Comment (Abhi2424shek) [inline docs/contributor-guide.md]: Can we have a branching policy here as well.

**Comment by AnaghHegde** [comment]
@Abhi2424shek @daminisatya Ready for review.

File: docs/contributor-guide.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+A self-upgrading process is available which must be triggered by a shell command. Just run:
+
+    > bin/upgrade.sh
+
+## Where can I download ready-built releases of Susi?
+
+No-where, you must clone the git repository of Susi and built it yourself. That's easy, just do
+- `git clone https://github.com/fossasia/susi_server.git`
+- `cd susi`
+- then see above ("How do I run Susi")
```
Comment (Abhi2424shek) [inline docs/contributor-guide.md]: Change this

File: docs/documentation-guide.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+
+## Where can I find documentation?
+
+The application has built-in documentation web pages, you will see them when you opened the application web pages or you can simply open `html/index.html` or just use http://api.asksusi.com as reference.
```
Comment (Abhi2424shek) [inline docs/documentation-guide.md]: dev.asksusi.org here pls

File: docs/documentation-guide.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+
+
+## Where can I find documentation?
+
+The application has built-in documentation web pages, you will see them when you opened the application web pages or you can simply open `html/index.html` or just use http://api.asksusi.com as reference. 
+
+## Where can I find showcases and tutorials?
+
+Articles and tutorials are also on our blog at http://blog.loklak.net.
```
Comment (Abhi2424shek) [inline docs/documentation-guide.md]: I think this one is changed

File: docs/documentation-guide.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+## Where do I find the javadocs?
+
+At http://susi.github.io/susi_server/ or by building them via 'ant javadoc'
```
Comment (Abhi2424shek) [inline docs/documentation-guide.md]: dev.asksusi.com/ something here

File: docs/contributor-guide.md (hunk 1 above)
Comment (AnaghHegde) [inline docs/contributor-guide.md]: I didn't get what you are trying to say.

File: docs/documentation-guide.md (hunk 4 above)
Comment (daminisatya) [inline docs/documentation-guide.md]: No this is still the same

File: docs/documentation-guide.md (hunk 3 above)
Comment (daminisatya) [inline docs/documentation-guide.md]: Do you think this URL is existing? can you check please

File: docs/contributor-guide.md (hunk 1 above)
Comment (Abhi2424shek) [inline docs/contributor-guide.md]: Well like what you asked we specify here that all development will go on the development branch and mention that the users need to commit to here. Something like that?

File: docs/documentation-guide.md (hunk 3 above)
Comment (AnaghHegde) [inline docs/documentation-guide.md]: It should be dev.asksusi.com

Comment (Abhi2424shek) [inline docs/documentation-guide.md]: Sorry my keyboard is dump it's dev.asksusi.com @daminisatya

File: docs/documentation-guide.md (hunk 4 above)
Comment (Abhi2424shek) [inline docs/documentation-guide.md]: K then, I mentioned as I read an issue where @mariobehling requested to seperate loklak components.

File: docs/contributor-guide.md (hunk 1 above)
Comment (daminisatya) [inline docs/contributor-guide.md]: There is a lot to be added. Along with the branch policy, I would like to add the coding practices too. Let me summarize

File: docs/documentation-guide.md (hunk 4 above)
Comment (daminisatya) [inline docs/documentation-guide.md]: This blog contains all the articles related to loklak as well susi too.

Comment (Abhi2424shek) [inline docs/documentation-guide.md]: 👍

File: docs/contributor-guide.md (hunk 1 above)
Comment (AnaghHegde) [inline docs/contributor-guide.md]: oh ok you are saying to add this `The issue has to be solved in development branch only and squash your commit into one after your done`. Correct me if i am wrong.

Comment (Abhi2424shek) [inline docs/contributor-guide.md]: @daminisatya then we will leave this PR off from such topics then :) open a new issue pls :)

Comment (Abhi2424shek) [inline docs/contributor-guide.md]: @AnaghHegde wait for @daminisatya

File: docs/documentation-guide.md (hunk 4 above)
Comment (AnaghHegde) [inline docs/documentation-guide.md]: @Abhi2424shek I m solving issue #6 also.So i think i need to change it or shall i submit a new PR for that after this.

File: docs/contributor-guide.md (hunk 1 above)
Comment (daminisatya) [inline docs/contributor-guide.md]: @Abhi2424shek @AnaghHegde Cool let's track it as a separate PR.

File: docs/documentation-guide.md (hunk 4 above)
Comment (daminisatya) [inline docs/documentation-guide.md]: This change requires some testing for accuracy. Lets track it differently. Open a different PR for it

**Comment by AnaghHegde** [comment]
@Abhi2424shek `dev.asksusi.com` contains the same `README.md` file information.So whats the point in again giving it as a reference in these readme file ?.

**Comment by Abhi2424shek** [comment]
Well either IMO remove that reference api.asksusi.com doesnot qualify for the reference, you may redirect them to http://dev.asksusi.com/docs/AskSUSI is a better reference but the .md file isnt being parsed by gh-pages!

File: docs/contributor-guide.md (hunk 2 above)
Comment (Abhi2424shek) [inline docs/contributor-guide.md]: Change this from above to below :)

**Comment by Abhi2424shek** [comment]
Squash your commits as well.

**Comment by AnaghHegde** [comment]
@Abhi2424shek How about if we add this reference after refactoring the README.md file ?and i did squash the commits.

**Comment by Abhi2424shek** [comment]
I can see 7 commits, ping on slack or gitter for help :) @AnaghHedge cool then just do as suggested change an above to below and squash and you are good enough to go.

**Comment by AnaghHegde** [comment]
@Abhi2424shek ok

**Comment by AnaghHegde** [comment]
@Abhi2424shek @daminisatya Ready for review.

**Comment by daminisatya** [comment]
@AnaghHegde thank you :)

**Pull request merged**